src/pages/publication/contributors_tab/ContributorArrows.tsx

```diff
@@ -18,18 +18,20 @@ const StyledDownArrow = styled.span`
 `;
 
 interface ContributorArrowsProps {
-  contributor: ContributorType;
-  onMoveContributor: (contributor: ContributorType, direction: number) => void;
+  index: number;
+  first: boolean;
+  last: boolean;
+  swap: (indexA: number, indexB: number) => void;
 }
 
-const ContributorArrows: React.FC<ContributorArrowsProps> = ({ contributor, onMoveContributor }) => {
+const ContributorArrows: React.FC<ContributorArrowsProps> = ({ first, last, swap, index }) => {
   return (
     <StyledContributorArrows>
       <StyledUpArrow>
-        <ArrowDropUpIcon onClick={() => onMoveContributor(contributor, Direction.ARROW_UP)} />
+        {(!first && <ArrowDropUpIcon onClick={() => swap(index, index - 1)} />) || <StopIcon />}
       </StyledUpArrow>
       <StyledDownArrow>
-        <ArrowDropDownIcon onClick={() => onMoveContributor(contributor, Direction.ARROW_DOWN)} />
+        {(!last && <ArrowDropDownIcon onClick={() => swap(index, index + 1)} />) || <StopIcon />}
       </StyledDownArrow>
     </StyledContributorArrows>
   );
```

Comment:
jeg synes denne filen er litt undøvendig. vanskelig å skjønne hva som skjer i Contributor.tsx når man må via styling-siden først, for å finne ut at det er en egen komponent. jeg synes det hadde vært bedre å putte StyledUpArrow og StyledDownArrow direkte inn i Contributor.tsx. da slipper vi masse props som blir sendt unødvendig også

Reply:
Pilene brukes i OtherContributor og ContributorValidator også, men må uansett se på hvordan disse komponentene blir når det er bestemt hvordan det skal være